### PR TITLE
travis build

### PR DESCRIPTION
**Pull request by rotty3000**

### CODE REVIEW

**Comment by rotty3000** [comment]
@waynebeaton if you or someone with better knowledge of the eclipse infra can take a quick look to see if this is suitable to bootstrap the travis build I'd appreciate it.

**Comment by m0mus** [comment]
It looks good to me. I assume that you tested it and it works.

**Comment by m0mus** [comment]
@lukasj can you take a look on this?

**Comment by rotty3000** [comment]
@m0mus how am I supposed to test this to make sure it works I cannot myself make this integration work. We need some help to assert this.

**Comment by lukasj** [comment]
Once you connect your account (rotty3000) to travis, travis should automagically run builds from your travis-enabled branch.

**Comment by rotty3000** [comment]
@lukasj my account is already in travis!

**Comment by rotty3000** [comment]
I had to manually enable this repo in my jenkins profile!

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+jdk:
+  - oraclejdk8
+  - oraclejdk9
+  - oraclejdk11
+  - openjdk8
+  - openjdk9
+  - openjdk10
+  - openjdk11
+  - openjdk12
```
Comment (lukasj) [inline .travis.yml]: Are all these JDKs really required? Shouldn't we focus only on the most important (LTS) ones - those would be 8, 11  and maybe latest available non-LTS (12)?

Keep in mind that in the end, number of executors available to all projects in eclipse-ee4j org is limited.

Comment (rotty3000) [reply .travis.yml]: reasonable. what about:
```
  - oraclejdk8
  - oraclejdk11
  - openjdk8
  - openjdk11
```

Comment (lukasj) [reply .travis.yml]: this list looks better to me

**Comment by rotty3000** [comment]
It looks like javadoc doclint is failing on java 11.

**Comment by rotty3000** [comment]
solution to javadoc issue https://bugs.openjdk.java.net/browse/JDK-8212233

You can see the passing build here https://travis-ci.org/rotty3000/common-annotations-api

**Comment by rotty3000** [comment]
@lukasj is this fine now?

**Comment by lukasj** [comment]
alternative solution to the javadoc build problem is to [turn of detectApiLink](https://maven.apache.org/plugins/maven-javadoc-plugin/javadoc-mojo.html#detectJavaApiLink) - not sure which one is better. I'm generally fine with whatever works in this particular case

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+cache:
+  directories:
+    - $HOME/.m2
+
+install:
+  - mvn install -DskipTests=true -Dmaven.javadoc.skip=true -B -V
+
+script:
+  - mvn verify -B -V
```
Comment (lukasj) [inline .travis.yml]: is build (in install) and rebuild (in script) really needed? I believe doing it once is enough

**Comment by rotty3000** [comment]
trying both suggestions! let's see...

**Comment by rotty3000** [comment]
Ok @lukasj everything is good with your suggestions https://travis-ci.org/rotty3000/common-annotations-api

**Comment by lukasj** [comment]
@m0mus feel free to merge this - I'm not a committer in this project...

**Comment by rotty3000** [comment]
Can I get anyone to bless this please?

**Comment by rotty3000** [comment]
@m0mus this merge will break the travis build because the spec has an enforced limitation to build only on Java 8.

**Comment by rotty3000** [comment]
Should I remove the upper bound?

**Pull request merged**